lib/runner/request-helpers-presend.js

```diff
@@ -41,32 +43,35 @@ module.exports = [
             authType = auth.type,
             authHandler = AuthLoader.getHandler(authType),
             authPreHook,
+            authInterface,
             authSignHook = function () {
                 try {
-                    authHandler.sign(auth, context.item.request, function (err) { done(err); });
+                    authHandler.sign(authInterface, context.item.request, function (err) { done(err); });
```

Comment:
Sign is in a try block while pre post and init are not. That's okay for now?

Reply:
The reason, why `sign` hook in a `try catch` is because

1. In the old implementation sign was the only synchronous API
2. We are using third party libraries mostly in sign hooks, which may not strictly adhere to sending error objects in callbacks.

Going back, we use third-party in other hooks as well.

Should we move other callbacks now, or wait till custom auth comes in?

Reply:
NTLM is already using 3rd party module in its post hook, so we should do this.
@shamasis should I do this in a separate PR?